src/block/mod.rs

```diff
@@ -515,15 +515,16 @@ impl UncompressedBlock {
         let absolute_indices = header.get_absolute_block_pixel_coordinates(tile_coordinates)?;
         absolute_indices.validate(Some(header.layer_size))?;
 
-        debug_assert_eq!(
+        if !header.compression.may_loose_data() { debug_assert_eq!(
```

Comment:
This cannot use `similar_to_lossy`, because this code does not know whether these bytes are f16, f32, u32, or any combination of these. Being aware of that would be very complicated for a simple check.

Reply:
Make sense.